File: Examples/CSharp/ProgrammersGuide/Presentations/RemoveWriteProtection.cs
//////////////////////////////////////////////////////////////////////////
// Copyright 2001-2015 Aspose Pty Ltd. All Rights Reserved.
//
// This file is part of Aspose.Slides. The source code in this file
// is only intended as a supplement to the documentation, and is provided
// "as is", without warranty of any kind, either expressed or implied.
//////////////////////////////////////////////////////////////////////////
using System.IO;

using Aspose.Slides;

namespace CSharp.Presentations
{
    public class RemoveWriteProtection
    {
        public static void Run()
        {
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_Presentations();

            //Opening the presentation file
            Presentation presentation = new Presentation(dataDir + "RemoveWriteProtection.pptx");

            //Checking if presentation is write protected
            if (presentation.ProtectionManager.IsWriteProtected)
                //Removing Write protection                
                presentation.ProtectionManager.RemoveWriteProtection();

            //Saving presentation
            presentation.Save(dataDir + "File_Without_WriteProtection.pptx", Aspose.Slides.Export.SaveFormat.Pptx);
        }
    }
}

File: Examples/VisualBasic/ProgrammersGuide/Presentations/RemoveWriteProtection.vb
'////////////////////////////////////////////////////////////////////////
' Copyright 2001-2013 Aspose Pty Ltd. All Rights Reserved.
'
' This file is part of Aspose.Slides. The source code in this file
' is only intended as a supplement to the documentation, and is provided
' "as is", without warranty of any kind, either expressed or implied.
'////////////////////////////////////////////////////////////////////////

Imports Microsoft.VisualBasic
Imports System.IO

Imports Aspose.Slides

Namespace VisualBasic.Presentations
    Public Class RemoveWriteProtection
        Public Shared Sub Run()
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_Presentations()

            'Opening the presentation file
            Dim pres As New Presentation(dataDir & "RemoveWriteProtection.pptx")


            'Checking if presentation is write protected
            If pres.ProtectionManager.IsWriteProtected Then
                'Removing Write protection
                pres.ProtectionManager.RemoveWriteProtection()
            End If

            'Saving presentation
            pres.Save(dataDir & "newDemo.pptx", Aspose.Slides.Export.SaveFormat.Pptx)
        End Sub
    End Class
End Namespace